### PR TITLE
[GEARPUMP-268] Accept java.time.Instant in Message

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [x] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

**Comment by manuzhang** [comment]
R: @huafengw

File: examples/streaming/state/src/main/scala/org/apache/gearpump/streaming/examples/state/processor/CountProcessor.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ class CountProcessor(taskContext: TaskContext, conf: UserConfig)
   }
 
   override def processMessage(state: PersistentState[Int], message: Message): Unit = {
-    state.update(message.timestamp, 1)
+    state.update(message.timestamp.toEpochMilli, 1)
```
Comment (huafengw) [inline examples/streaming/state/src/main/scala/org/apache/gearpump/streaming/examples/state/processor/CountProcessor.scala]: message.timestamp.toEpochMilli == message.timeInMillis

Comment (manuzhang) [inline examples/streaming/state/src/main/scala/org/apache/gearpump/streaming/examples/state/processor/CountProcessor.scala]: yes, but I propose to hide `timeInMillis` from users in the future.

Comment (manuzhang) [inline examples/streaming/state/src/main/scala/org/apache/gearpump/streaming/examples/state/processor/CountProcessor.scala]: It's only used internally in `Subscription` and can be replaced in the future as well.

**Comment by manuzhang** [comment]
travis has passed but not synced here somehow. Please check https://travis-ci.org/apache/incubator-gearpump/builds/199856894. Also, code coverage only runs when travis has passed.

**Comment by huafengw** [comment]
+1

**Pull request closed without merging**